nyaa/api_torznab.py

```diff
@@ -101,71 +112,113 @@ def torznab_caps():
     return render_torznab_caps()
 
 
-def torznab_search():
-    page_size = app.config['RESULTS_PER_PAGE']
+def get_categories_filter(cat):
+    categories = []
+    if cat:
+        for item in cat.split(','):
+            # Parse 1_1 format for compatibility
+            cat_match = re.match(r'^(\d+)_(\d+)$', item)
+            if cat_match:
+                item = int(cat_match.group(1)) * 100 + int(cat_match.group(2))
+
+            num = int(item)
+            if num >= 100000:
+                categories.append(num - 100000)
+            elif num >= 1000:
+                categories.extend([x - 100000 for x in TORZNAB_CATEGORY_MAP.get(num, [])])
+            elif num > 0:
+                categories.append(num)
+
+        # Check if any of the specified categories 'stuck'.
+        if not categories:
+            raise EmptySearchError('Categories {} guaranteed to produce empty result'.format(cat))
+    return categories
+
+def get_tag_filter(tag):
+    include_tags = []
+    exclude_tags = []
+    if tag:
+        for item in tag.split(','):
+            if item[0] == '!':
+                exclude_tags.append(item[1:].lower())
+            else:
+                include_tags.append(item.lower())
+    for incl in include_tags:
```

Comment:
If you use sets instead of lists for 'include_tags' and 'exclude_tags'  this becomes a simple set operation instead of another iteration.
```python
if included.isdisjoint(excluded):
```